connectors/sources/sharepoint_online.py

```diff
@@ -910,8 +910,10 @@ def __init__(self, configuration):
 
         if self.configuration["use_text_extraction_service"]:
             self.extraction_service = ExtractionService()
+            self.download_dir = self.extraction_service.get_volume_dir()
         else:
             self.extraction_service = None
+            self.download_dir = None
```

Comment:
This looked concerning, but after a little digging found in https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp:

> If dir is not None, the file will be created in that directory; otherwise, a default directory is used.

Leaving a comment here in case this startles other reviewers.